### PR TITLE
Add cops for resource / property descriptions

### PR DESCRIPTION
**Pull request by tas50**
These are off by default, but allow us to enforce descriptions in our repos.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: config/cookstyle.yml [hunk 1]
```diff
@@ -386,6 +386,21 @@ ChefSharing/EmptyMetadataField:
...
+  Include:
+    - '**/libraries/*.rb'
+    - '**/resources/*.rb'
+
+ChefSharing/IncludeResourceDescriptions:
+  Description: Resources should include description fields to allow automated documention. Requires Chef Infra Client 13.9 or later.
+  Enabled: false
+  VersionAdded: '6.1.0'
+  Include:
+    - '**/resources/*.rb'
```
Comment (phiggins) [inline config/cookstyle.yml]: Was leaving `    - '**/libraries/*.rb'` off of this one intentional?

Comment (tas50) [reply config/cookstyle.yml]: Yeah things are trickier there. I was leaving it out for now

File: lib/rubocop/cop/chef/sharing/include_property_descriptions.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+module RuboCop
+  module Cop
+    module Chef
+      module ChefSharing
+        # Resources properties should include description fields to allow automated documention. Requires Chef Infra Client 13.9 or later.
```
Comment (phiggins) [inline lib/rubocop/cop/chef/sharing/include_property_descriptions.rb]:
```suggestion
        # Resource properties should include description fields to allow automated documention. Requires Chef Infra Client 13.9 or later.
```

Should this be plural?

File: lib/rubocop/cop/chef/sharing/include_property_descriptions.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        #
+        #   # good
+        #   property :foo, String, description: "Set the important thing to..."
+        #
+        class IncludePropertyDescriptions < Cop
+          extend TargetChefVersion
+
+          minimum_target_chef_version '13.9'
+
+          MSG = 'Resources should include description fields to allow automated documention. Requires Chef Infra Client 13.9 or later.'.freeze
```
Comment (phiggins) [inline lib/rubocop/cop/chef/sharing/include_property_descriptions.rb]:
```suggestion
          MSG = 'Resource properties should include description fields to allow automated documention. Requires Chef Infra Client 13.9 or later.'.freeze
```

**Pull request merged**